### ISSUE TITLE
libgd-2.2.3: Tests bug00201 and 00067 fail on non-x86_64 architectures

### ISSUE THREAD

**Issue by badshah400** [libgd/libgd#313]
Building libgd-2.2.3 on i586 or any other non-x86_64 architectures leads to two failing tests:

```
====================================
   GD 2.2.3: tests/test-suite.log
====================================

# TOTAL: 125
# PASS:  123
# SKIP:  0
# XFAIL: 0
# FAIL:  2
# XPASS: 0
# ERROR: 0

.. contents:: :depth: 2

FAIL: gdimagecopyresampled/bug00201
===================================

gdimagecopyresampled/bug00201.c:68: Total pixels changed: 4256 with a maximum channel difference of 255.
FAIL gdimagecopyresampled/bug00201 (exit status: 1)

FAIL: gdimagerotate/bug00067
============================

gdimagerotate/bug00067.c:46: Total pixels changed: 14143 with a maximum channel difference of 255.
gdimagerotate/bug00067.c:47: comparing rotated image to ./gdimagerotate/bug00067_150_exp.png failed.
FAIL gdimagerotate/bug00067 (exit status: 1)
```

I am trying to build against freetype 2.6.5. I will be all too happy to get back with any other info. Thanks!

**Comment by cmb69** [comment]
Please try vapier' suggestion from <https://github.com/libgd/libgd/issues/242#issuecomment-228676965>.

**Comment by vapier** [comment]
by "non-x86_64", do you really just mean "32-bit x86" ?  have you actually tested other arches like arm or mips or s390 or ppc ?

**Comment by badshah400** [comment]
>  by "non-x86_64", do you really just mean "32-bit x86" ? have you actually tested other arches like arm or mips or s390 or ppc ?

These are the archs on which it fails:

* openSUSE_Factory_PowerPC ppc64le    failed
* openSUSE_Factory_PowerPC ppc64      failed
* openSUSE_Factory_PowerPC ppc        failed
* openSUSE_Factory_ARM armv7l     failed
* openSUSE_Factory_ARM aarch64    failed

**Comment by badshah400** [comment]
@cmb69 @vapier 
Adding "-msse -mfpmath=sse" seems to have helped (at least on i586). I will update my report on builds for other archs soon. Thanks!

**Comment by badshah400** [comment]
Right, so still does not build with all the archs mentioned in my previous comment. But it works for i586

**Comment by cmb69** [comment]
Thanks for testing. I presume we're facing precision/rouding issues here. Can you please post the actual images and the diffs?

**Comment by badshah400** [comment]
Sorry, I should have checked more carefully: the reason for the build for these arch failing is not any test, but entirely different. The i586 was failing because of test failures, but it is fixed with the additional cflags. This bug is closed for me. Thanks for your help.

FYI, the builds for the other archs fail with messages similar to (but all of them used to build with version 2.1.1, which was the previous version we had pkged for openSUSE [1]):

```
[   52s] checking for powerpc64le-suse-linux-gnu-gcc... no
[   52s] checking for gcc... gcc
[   52s] checking whether the C compiler works... no
[   52s] configure: error: in `/home/abuild/rpmbuild/BUILD/libgd-2.2.3':
[   52s] configure: error: C compiler cannot create executables
```

[1] https://build.opensuse.org/package/show/home:badshah400:freetype27/gd2

**Comment by cmb69** [comment]
Okay, so I'm closing. Please report possible build problems as another issue. Thanks!

FTR: the unitialized variable error has already been fixed with 2b3dd57a.
